### PR TITLE
Update GEP overview to clarify the GEP process

### PR DESCRIPTION
**Pull request by youngnick**
/kind documentation

**What this PR does / why we need it**:
This updates the GEP overview page to add a state diagram and clarify some steps.'

Also includes cleaning out the `site-src/geps` directory because I was having problems getting it to update properly.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Review by shaneutt — APPROVED** [review]
I think the overview of what the process should roughly be is good, and then specifically asking for GEP issues is good too, so we can better track these on our project boards and for milestones.

/lgtm

I think this should be pretty non-controversial, but I'll put a hold so we can have a couple other people give it a glance before we merge it.

/hold

File: geps/overview.md
Comment (mlavacca) [inline geps/overview.md:25]: Since the Gateway API does not ship controllers, what does "Implemented in controllers" mean? Maybe we can say something like "implemented in the API"?

Comment (youngnick) [reply geps/overview.md:25]: The idea here is that things aren't moved to standard until enough implementations have finished working on them. "Implemented in implementations" didn't sound good, but maybe "Supported in multiple implementations" is a better way of phrasing that will make it clearer and more standard.

Comment (youngnick) [reply geps/overview.md:25]: I updated, and added a note about conformance tests.

**Review by mlavacca — APPROVED** [review]
/lgtm

**Pull request merged**
